.github/dependabot.yml

```diff
@@ -12,7 +12,19 @@ updates:
     allow:
       - dependency-type: "all"
     ignore:
-      - dependency-name: "*pytest*", "flake8", "black", "mypy", "*sphinx*", "myst*", "jupytext"
+      - dependency-name: "*pytest*"
+        update-types: ["version-update:semver-minor", "version-update:semver-patch"]
+      - dependency-name: "flake8"
+        update-types: ["version-update:semver-minor", "version-update:semver-patch"]
+      - dependency-name: "black"
```

Comment:
Is there something weird about Black such that we shouldn't update it for major versions?

Reply:
For formatting it seems like just sticking to one version is best so that it's always the same code format. Something happened in #886 for example that caused flake8 to fail. I wanted to avoid things like this.